### DISCUSSION TITLE
Using Laravel CustomRequest on API Resource

### DISCUSSION THREAD

**Discussion by ZakaCoding** [laravel/framework#53653] [Q&A]
### Laravel Version

11.*

### PHP Version

8.*

### Database Driver & Version

PostgreSQL

### Description

Hello.

i got a problem when using Validator for incoming request. When i use CustomRequest and i test with error, laravel always redirect with Welcome Page, not json response what i want. this is weird.

Is Laravel CustomRequest is only good with WEB Resource ?

Check this code for example.

`
 public function show(SomeModel $type, Request $request)
    {
        $validator = Validator::make($request->all(), [
            'uuid' => ['required'],
        ]);

        if($validator->fails())
        {
            return response([
                'meta' => [
                    'code' => 422,
                    'status' => 'error',
                    'message' => null,
                ],
                'results' => $validator->errors()
            ], 422);
        }

       // Just sent POST Request without uuid, it should be return JSON 422 right ?
       // But it will return page of laravel in POSTMAN.
    }`

### Steps To Reproduce

i dont know about this :)

**Comment by AmadulHaque** [comment]
You can use this Example code:

<?php

namespace App\Http\Requests;

use Illuminate\Http\JsonResponse;
use Illuminate\Foundation\Http\FormRequest;
use Illuminate\Http\Exceptions\HttpResponseException;
use Illuminate\Contracts\Validation\Validator;
class ExampleRequest extends FormRequest
{
    /**
     * Determine if the user is authorized to make this request.
     */
    public function authorize(): bool
    {
        return true; // or false if you want
    }

    /**
     * Get the validation rules that apply to the request.
     *
     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
     */
    public function rules(): array
    {
        return [
            'name'   => 'required',
        ];
    }

    public function failedValidation(Validator $validator) : JsonResponse
    {
        $errors = $validator->errors();
        throw new HttpResponseException(validationError('Validation Error',  errors: $errors));
    }

}

**Comment by ZakaCoding** [comment]
> failedValidation(

i have use this before. and still doesn't work

**Reply by ghost** [reply]
What Accept header are you sending in your request?
